x-pack/plugins/security_solution/public/common/components/exceptions/add_exception_modal/index.tsx

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useEffect, useState, useCallback, useMemo } from 'react';
+import styled, { css } from 'styled-components';
+import {
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalFooter,
+  EuiOverlayMask,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiHorizontalRule,
+  EuiCheckbox,
+  EuiSpacer,
+  EuiFormRow,
+  EuiCallOut,
+  EuiText,
+} from '@elastic/eui';
+import { alertsIndexPattern } from '../../../../../common/endpoint/constants';
+import {
+  ExceptionListItemSchema,
+  ExceptionListSchema,
+  CreateExceptionListItemSchema,
+} from '../../../../../public/lists_plugin_deps';
+import * as i18n from './translations';
+import { TimelineNonEcsData, Ecs } from '../../../../graphql/types';
+import { useKibana, useUiSetting$ } from '../../../lib/kibana';
+import { errorToToaster, displaySuccessToast, useStateToaster } from '../../toasters';
+import { ExceptionBuilder } from '../builder';
+import { Loader } from '../../loader';
+import { useAddOrUpdateException } from '../../../../alerts/containers/detection_engine/alerts/use_add_exception';
+import { useSignalIndex } from '../../../../alerts/containers/detection_engine/alerts/use_signal_index';
+import { useFetchOrCreateRuleExceptionList } from '../use_fetch_or_create_rule_exception_list';
+import { AddExceptionComments } from '../add_exception_comments';
+import {
+  enrichExceptionItemsWithComments,
+  enrichExceptionItemsWithOS,
+  defaultEndpointExceptionItems,
+  entryHasListType,
+  entryHasNonEcsType,
+} from '../helpers';
+import { useFetchIndexPatterns } from '../../../../alerts/containers/detection_engine/rules';
+
+// TODO: move somewhere else?
+// TODO: rename?
+export interface AddExceptionOnClick {
+  ruleName: string;
+  ruleId: string;
+  ruleExceptionLists?: ExceptionListSchema[];
```

Comment:
I might be explicit here like `ruleExceptionLists: ExceptionListSchema[]  | undefined;` Because we're using it as a tell for whether the user is creating a new list vs. editing right?